docs/proposals/extracted_credentials_saved_as_secrets.md

```diff
@@ -0,0 +1,59 @@
+## Extracted Credentials Saved As Secrets
+
+Extracted Credentials are currently saved in our etcd for the service broker. This is not desirable for many reasons, but the two biggest are kubernetes already has a built-in way to manage this data, secrets, and when moving to CRDs we don't want to create a resource for extracted credentials.
+
+### Problem Description
+The problem is that we should not manage data that is of a sensitive nature if we do not have to. This proposal is limited in scope and only interested in how we save the extracted credentials. It is worth noting that we should eventually be better about how we transmit this data to APBs. 
+
+In the secret, we will save the data in the following format.
+```yaml
+data:
+  DB_PASSWORD: <GOB_ENCODED_VALUE>
+  DB_USERNAME: <GOB_ENCODED_VALUE>
+  ....
+apiVersion: v1
+kind: Secret
+metadata:
+  name: <Service/Binding id>
+  namespace: <Namespace for the broker>
+  labels:
+    <labels provided>
+```
+
+[Gob encoding](https://godoc.org/encoding/gob) will allow us to save arbitrary data in the secret for a key. The secrets keys will look rational to a user who looks at the created secret. This user would need permissions to see the secret, but if someone is looking at the secret making it obvious what data is in there will be helpful. 
```

Comment:
I'm not familiar with this pkg so excuse me if these should be obvious:

* Correct me if I'm wrong, but this is not required to accomplish what we're doing? What is the benefit, an independent storage format?
* We've talked about encrypting credentials, is that something that can be done easily here? (Or is that unnecessary thanks to our usage of secrets?)

Reply:
Additionally, is base64 encryption enough? Is that what the rest of openshift deems secure enough?

Reply:
We have a map[string]interface{}. The secrets data needs to be a`[]byte` transforming arbitrary data, `interface{}` to a byte array is tricky. This is one of the options that I found how to do this. I thought it would be helpful because the how to decode the value is included in the encoding so that even outside people could just use the built-in gob decoder for golang and decode the value. 

With that being said, we could just marshal the entire `map[string]interface` to json and only have 1 key in the secret called credentials. I didn't think this was as good because then when a user who is debugging stuff has to know how we store it and the above way they will just have encoded values but the keys for the extracted credentials will be at the top level of the secret. 

This is not encrypting the secret, but secrets are well understood and as far as I know, the correct place to put this data without encrypting it. So I was not planning on adding encryption here.

Reply:
+1 on both accounts. Thanks @shawn-hurley.